### ISSUE TITLE
Allow caching of query results

### ISSUE THREAD

**Issue by SamSaffron** [wrouesnel/postgres_exporter#210]
Getting pg database size is very expensive, especially if we need to do this across hundreds of dbs.

I would like an optional `cache` property per query that allows us to "remember" the last execution of the query for N seconds... Eg:  

```
pg_database:
  query: " SELECT pg_database.datname, pg_database_size(pg_database.datname) as size FROM pg_database" 
  cache: 3600
  metrics:
    - datname:
        usage: "LABEL"
        description: "Name of the database"
    - size:
        usage: "GAUGE"
        description: "Disk space used by the database"
```

Is there any objection to adding such an optional attribute. 

Currently we have no way of reporting DB size cause if we report it we hammer our DBs into oblivion

**Comment by baryluk** [comment]
+1, however I also think pg_database_size should be included in default set of metrics anyway ;) with suitable caching.

**Comment by wrouesnel** [comment]
Resolved by #319 which was a rewrite of #211 since quite a lot had changed. Hopefully working v0.7.0 which I just pushed to get these features out (latest will have them too).
